lib/Pool.js

```diff
@@ -105,6 +106,29 @@ class Pool {
       }
     });
 
+    // Fires when a connection has errored. If there is no work waiting, then
+    // the pool will simply be empty until the next time work is requested, at
+    // which point there WILL be waiting work and they will get an error.
+    this.connectionEmitter.on('connectionError', (error) => {
+      if (this.waitingConnectionWork.length > 0)
+      {
+        let connectionWork = this.waitingConnectionWork.shift();
```

Comment:
Could always call shift() and check if connectionWork is undefined when empty instead.

Reply:
Good call, I fixed in the "connected" event as well (the mismatch would drive me crazy otherwise)